### PR TITLE
refactor reading state and config data

### PR DESCRIPTION
**Pull request by rh-ulrich-o**
This pull request consists of three commits:
1. introduce the getJSONstring() function
2. call getJSONstring() to read state data from standard input
3. call getJSONstring() to read config data from bundlePath/config.json

### CODE REVIEW

File: src/systemdhook.c [hunk 1]
```diff
@@ -728,23 +728,19 @@ int main(int argc, char *argv[])
...
-	rd = fread((void *)stateData, 1, sizeof(stateData) - 1, stdin);
-	if (rd == 0 && !feof(stdin)) {
-		pr_perror("Error encountered on file read");
-		return EXIT_FAILURE;
-	} else if (rd >= sizeof(stateData) - 1) {
-		pr_perror("Config file too big");
+	/* Read the entire state from stdin */
+	stateData = getJSONstring(stdin, (size_t)CHUNKSIZE, "state data");
+	if (stateData == NULL) {
+		pr_perror("failed to read state data from standard input");
```
Comment (rhatdan) [inline src/systemdhook.c]: Why not just pass this message into the getJSONstring msg field?  Then you would just get one error message in the log not two.

File: src/systemdhook.c [hunk 2]
```diff
@@ -793,38 +790,19 @@ int main(int argc, char *argv[])
...
-		return EXIT_FAILURE;
-	}
-	configData_size = configData_stat.st_size;
-
-	configData = (char *)malloc((size_t)configData_size+1);
+	/* Read the entire config file */
+	configData = getJSONstring(fp, (size_t)CHUNKSIZE, "config data");
 	if (configData == NULL) {
-		pr_perror("Failed to allocate buffer for config data: size=%ld", configData_size+1);
+		pr_perror("failed to read config data from %s", config_file_name);
```
Comment (rhatdan) [inline src/systemdhook.c]: Why not just pass this message into the getJSONstring msg field? Then you would just get one error message in the log not two.

Then just return EXIT_FAILURE.

**Comment by rhatdan** [comment]
Verify nice patch.

@mrunalp PTAL

File: src/systemdhook.c (hunk 1 above)
Comment (rh-ulrich-o) [inline src/systemdhook.c]: I implemented the suggested change in commit d3ae647b8dbc4949c702fe60cc0605f63fc849eb that I added to this pull request / topic branch today.
"avoid multiple log messages if getJSONstring() fails"

File: src/systemdhook.c (hunk 2 above)
Comment (rh-ulrich-o) [inline src/systemdhook.c]: I implemented the suggested change in commit d3ae647b8dbc4949c702fe60cc0605f63fc849eb
that I added to this pull request / topic branch today.
"avoid multiple log messages if getJSONstring() fails"

**Comment by rh-ulrich-o** [comment]
Added three commits to this pull request:

1. address a minor issue in getJSONstring() that I found myself
2. implement changes suggested by Dan
3. fix a few comment lines

Changes in this pull request / topic branch were tested as shown in the attachment.
[test.txt](https://github.com/projectatomic/oci-systemd-hook/files/949814/test.txt)

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**